### PR TITLE
Introduce configuration tolerance

### PR DESCRIPTION
**Pull request by paveltomin**
Introduce configuration tolerance, the configuration is assumed converged if changed area is below the defined fraction of total area:
`changedArea <= configurationTolerance * totalArea`

Limits for `configurationTolerance`:
- minimum value is 0 (default) when any configuration change makes it unconverged
- values >=1 mean effectively configuration check is disabled

No default behavior change but needs rebaseline because of the new parameter.

### CODE REVIEW

File: src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp [hunk 1]
```diff
@@ -1941,27 +1944,28 @@ bool SolidMechanicsLagrangeContact::updateConfiguration( DomainPartition & domai
...
-                         MPI_LAND,
-                         MPI_COMM_GEOSX );
+  // Compute global area of changed elements
+  changedArea = MpiWrapper::sum( changedArea );
+  // and total area of fracture elements
+  totalArea = MpiWrapper::sum( totalArea );
 
-  return hasConfigurationConvergedGlobally;
+  // Assume converged if changed area is below certain fraction of total area
+  return changedArea <= m_nonlinearSolverParameters.m_configurationTolerance * totalArea;
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp:1968]: does this mean that if only a small element changes state we do not update the configuration?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp:1968]: My intention was to allow to converge when only a small portion of fracture elements change state
Typically happens when things are flip flopping

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp:1968]: Configuration should be updated, this PR should be only about convergence criteria for configuration change - currently it is strict yes or no, here we are trying to allow some tolerance

**Comment by paveltomin** [comment]
please review

**Review by matteofrigo5 — APPROVED** [review]
It looks fine to me

**Review by jhuang2601 — APPROVED** [review]

**Review by Guotong-Ren — APPROVED** [review]

**Pull request merged**